drivers/SmartThings/matter-switch/src/aqara-cube/init.lua

```diff
@@ -0,0 +1,216 @@
+      device.log.debug("configuring for press event only")
+      set_field_for_endpoint(device, INITIAL_PRESS_ONLY, ep, true, true)
+    end
+  end
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+    local MS = get_reordered_endpoints(driver, device)
+    local main_endpoint = device.MATTER_DEFAULT_ENDPOINT
+    if #MS > 0 then
+      main_endpoint = MS[1] -- the endpoint matching to the non-child device
+      if MS[1] == 0 then main_endpoint = MS[2] end -- we shouldn't hit this, but just in case
+    end
+    device.log.debug("main button endpoint is "..main_endpoint)
+
+    -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
+    local current_component_number = 1
+    local component_map = {}
+    for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
+      device.log.debug("Configuring endpoint "..ep)
+      -- build the mapping of endpoints to components
+      component_map[string.format("%d", current_component_number)] = ep
+      current_component_number = current_component_number + 1
+    end
+
+    device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
+    device:try_update_metadata({profile = "cube-t1-pro"})
+    device:set_field(DEFERRED_CONFIGURE, true)
+  end
+end
+
+local function info_changed(driver, device, event, args)
+  if device.profile.id ~= args.old_st_store.profile.id
+    and device:get_field(DEFERRED_CONFIGURE)
+    and device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+
+    -- At the time of device_added, there is an error that the corresponding capability cannot be found
+    -- because the profile has not been changed from the generic profile of fingerprint to the Aqara Cube.
+    reset_thread(device)
+    device:emit_event(cubeFace.cubeFace("face1Up"))
+
+    -- profile has changed, and we deferred setting up our buttons, so do that now
+    configure_buttons(device)
+    device:set_field(DEFERRED_CONFIGURE, nil)
+  end
+end
+
+-- initial press
+local function initial_press_event_handler(driver, device, ib, response)
+  if get_field_for_endpoint(device, INITIAL_PRESS_ONLY, ib.endpoint_id) then
+    local map = device:get_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON) or {}
+    local face = 1
+    for component, ep in pairs(map) do
+      if map[component] == ib.endpoint_id then
+        face = component
+        break
+      end
+    end
+
+    reset_thread(device)
+    device:emit_event(cubeAction.cubeAction(string.format("flipToSide%d", face)))
+    device:emit_event(cubeFace.cubeFace(string.format("face%dUp", face)))
+  end
+end
+
+local function battery_percent_remaining_attr_handler(driver, device, ib, response)
+  if ib.data.value then
+    device:emit_event(capabilities.battery.battery(math.floor(ib.data.value / 2.0 + 0.5)))
+  end
+end
+
+-- [[ register ]]
+local aqara_cube_t1_pro_handler = {
+  NAME = "Aqara Cube T1 Pro",
```

Comment:
since the function `is_aqara_cube` is so general, as well as the aqara-cube subdriver name, should these also be more general?
```suggestion
local aqara_cube_handler = {
  NAME = "Aqara Cube Handler",
```

Reply:
Okay, I changed it.